# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mp4/MP4Demuxer.cpp
# Commit: da557d1806dd
# Full Hash: da557d1806dd9e1e519e00a4481b2e075f722ae8
# Author: Butkovits Atila <abutkovits@mozilla.com>
# Date: 2023-01-27 22:54 +0200
# Regressor Bug: 1788119
# File Overlap Count: 1
# Overlapping Files: Cargo.lock
# Description:
#   Backed out 5 changesets (bug 1788119) for causing bustages complaining about AVIFDecodedData. CLOSED TREE
# ==============================================================================

diff -r d980bebc6d26 -r da557d1806dd dom/media/mp4/MP4Demuxer.cpp
--- a/dom/media/mp4/MP4Demuxer.cpp	Fri Jan 27 20:35:34 2023 +0000
+++ b/dom/media/mp4/MP4Demuxer.cpp	Fri Jan 27 22:54:08 2023 +0200
@@ -13,6 +13,7 @@
 #include "AnnexB.h"
 #include "BufferStream.h"
 #include "H264.h"
+#include "Index.h"
 #include "MP4Decoder.h"
 #include "MP4Metadata.h"
 #include "MoofParser.h"
@@ -22,7 +23,6 @@
 #include "mozilla/StaticPrefs_media.h"
 #include "mozilla/Telemetry.h"
 #include "nsPrintfCString.h"
-#include "SampleIterator.h"
 
 extern mozilla::LazyLogModule gMediaDemuxerLog;
 mozilla::LogModule* GetDemuxerLog() { return gMediaDemuxerLog; }
@@ -66,7 +66,7 @@
   RefPtr<MediaResource> mResource;
   RefPtr<ResourceStream> mStream;
   UniquePtr<TrackInfo> mInfo;
-  RefPtr<MP4SampleIndex> mIndex;
+  RefPtr<Index> mIndex;
   UniquePtr<SampleIterator> mIterator;
   Maybe<media::TimeUnit> mNextKeyframeTime;
   // Queued samples extracted by the demuxer, but not yet returned.
@@ -309,8 +309,7 @@
     : mResource(aResource),
       mStream(new ResourceStream(aResource)),
       mInfo(std::move(aInfo)),
-      mIndex(new MP4SampleIndex(aIndices, mStream, mInfo->mTrackId,
-                                mInfo->IsAudio())),
+      mIndex(new Index(aIndices, mStream, mInfo->mTrackId, mInfo->IsAudio())),
       mIterator(MakeUnique<SampleIterator>(mIndex)),
       mNeedReIndex(true) {
   EnsureUpToDateIndex();  // Force update of index